src/Simulation/Core/Operations/Operation.cs

```diff
@@ -77,6 +77,11 @@ public O Apply(I a)
                 this.Factory?.StartOperation(this, __dataIn(a));
                 __result__ = this.Body(a);
             }
+            catch( Exception e)
```

Comment:
Will the `fail` statement throw an Exception right now? Can we use that same exception instead of the `ExceptionDispatchInfo`?

Reply:
`ExceptionDispatchInfo` containst exception itself and relevant debugging information from the moment exception was thrown. So if you say wanted to say access the call-stack of the exception at the moment when it was thrown, you could do it. Original exceptions is available through [`ExceptionDispatchInfo.SourceException`](https://docs.microsoft.com/dotnet/api/system.runtime.exceptionservices.exceptiondispatchinfo.sourceexception) property. 

Fail statement throws exceptions the same way as before.